lib/Directives/SubDirective.php

```diff
@@ -30,11 +30,34 @@ final public function process(
         string $data,
         array $options
     ): void {
-        $subParser = $parser->getSubParser();
-
-        if ($node instanceof CodeNode) {
-            $document = $subParser->parseLocal($node->getValue());
+        /*
+         * A BlockNode indicates that the content was passed in "raw" and should
+         * be sub-parsed. This is the main use-case.
+         *
+         * However, there are some odd cases where a directive appears
+         * to "end"... and DocumentParser continues parsing the next
+         * section, for example as a ParagraphNode or ListNode. When
+         * that terminates, the original directive is THEN processed
+         * and called.
+         *
+         * A key example is the "class::" directive (see class-directive.rst test case).
+         * That is where it is legal to have a format like this:
+         *
+         *      .. class:: special-list
+         *
+         *      - Test list item 1.
+         *      - Test list item 2.
+         *
+         * Notice the 2 list items are NOT indented. This is legal, and ultimately
+         * those two items would be parsed as a ListNode and THEN passed to
+         * ClassDirective (which extends SubDirective) for processing.
+         */
+        if ($node instanceof BlockNode) {
+            $subParser = $parser->getSubParser();
+            $document  = $subParser->parseLocal($node->getValue());
```

Comment:
Let's get rid of the `$subParser` one-time variable, as it does not bring extra clarity over `getSubParser()`? 
```suggestion
            $document  = $parser->getSubParser()->parseLocal($node->getValue());
```